package/MDAnalysis/analysis/helix_analysis.py

```diff
@@ -461,8 +461,8 @@ def _conclude(self):
         # flatten?
         if len(self.atomgroups) == 1 and self._flatten:
             for name in attrnames + ['summary']:
-                attr = getattr(self.results, name)
-                setattr(self.results, name, attr[0])
+                attr = self.results[name]
+                self.results[name] = attr[0]
```

Comment:
Is this really
```python
self.results[name] = self.results[name][0]
```
?

Reply:
It is. I think this is an issue with the Results class. The below fails in Results.

```python
def __setitem__(self, key, item):
        self._validate_key(key)
        super().__setitem__(key, item)
        assert self[key] is getattr(self, key)
```

Reply:
See [this falsifying test](https://github.com/MDAnalysis/mdanalysis/pull/3281/commits/cb2c34edfebbc079ac530515474fe51dfd3fa44e) for an example.